Sources/FoundationEssentials/String/String+IO.swift

```diff
@@ -14,8 +14,38 @@
 internal import _ForSwiftFoundation
 #endif
 
+internal import _FoundationCShims
+
 fileprivate let stringEncodingAttributeName = "com.apple.TextEncoding"
 
+private struct ExtendingToUTF16Sequence<Base: Sequence<UInt8>> : Sequence {
```

Comment:
I found this Sequence adaptor pattern to have less than stellar performance in the past. It's probably good to get us started but we should note that it may be possible to optimize.

Reply:
Ah good to know, yeah this is probably worth a performance benchmark as a followup here